### PR TITLE
Added dependsOn section with it's test

### LINKED ISSUE #328: Task Dependency (V-1)
### Create a dependency section  on create task form

**Description**
- [ ] Have to create a dependency section on the task form when a task is created. 
- [ ] Have to give task-id in the dependency section.

**Requirements**

- Dependency section takes the task-id to have a dependency on another task.
- TDD is to be followed

**UI**
RDS-dependsOn

### PR DESCRIPTION
**Pull request by vinit717**
closes #328

### CODE REVIEW

File: .yarn/install-state.gz
Comment (RitikJaiswal75) [inline .yarn/install-state.gz:1]: this file is not a part of PR, so why is it showing up?

File: yarn.lock
Comment (RitikJaiswal75) [inline yarn.lock:1]: this file is not a part of PR, so why is it showing up?

File: jest.config.js
Comment (RitikJaiswal75) [inline jest.config.js:1]: please remove the unused commented code

File: __test__/task.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          required
+          class="input notEditing"
+          id="dependsOn"
+          name="dependsOn"
+          multiple
+        >
+          <option disabled selected>Select dependsOn</option>
+        </select>
+      </div>
+    `);
```
Comment (RitikJaiswal75) [inline __test__/task.test.js:26]: Nice way to isolate the specific code and test it. 😍

File: __test__/task.test.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  afterAll(async () => {
+    await browser.close();
+  });
+
+  test('fetches tasks from API', async () => {
+    await page.goto(`${API_BASE_URL}/tasks`);
+    const response = await page.evaluate((baseUrl) => {
+      return fetch(`${baseUrl}/tasks`).then((response) => response.json());
+    }, API_BASE_URL);
```
Comment (RitikJaiswal75) [inline __test__/task.test.js:67]: why do we have API_BASE URL_HERE?

File: task/script.js [hunk 3]
```diff
@@ -220,7 +222,7 @@ taskForm.onsubmit = async (e) => {
   } else {
     delete dataToBeSent.links;
   }
-
+  console.log(dataToBeSent);
```
Comment (RitikJaiswal75) [inline task/script.js:225]: do we need this console log in production?

File: jest.config.js
Comment (vinit717) [reply jest.config.js:1]: it is auto generated  by jest --init so should I have to remove it ?

File: task/script.js (hunk 3 above)
Comment (vinit717) [reply task/script.js]: removed

File: __test__/task.test.js (hunk 2 above)
Comment (vinit717) [reply __test__/task.test.js:67]: to fetch the response as an alternative solution causes an error in importing the function

File: .yarn/install-state.gz
Comment (vinit717) [reply .yarn/install-state.gz:1]: removed ?

File: yarn.lock
Comment (vinit717) [reply yarn.lock:1]: when I install jest and jsdom it automatically changed so should I revert the changes?

**Pull request closed without merging**